staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -576,21 +576,21 @@ func TestReflectorFullListIfExpired(t *testing.T) {
 			case "0":
 				return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil
 			case "10":
+				if options.Continue != "" {
+					t.Fatalf("specifying resource version (%s) is not allowed when using continue (%s).", options.ResourceVersion, options.Continue)
+				}
 				switch options.Limit {
 				case 4:
-					switch options.Continue {
-					case "":
-						return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
-					case "C1":
-						return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
-					default:
-						t.Fatalf("Unrecognized Continue: %s", options.Continue)
-					}
-				case 0:
+					return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
+				case 0: // Full re-list
 					return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "11"}, Items: pods[0:8]}, nil
 				default:
 					t.Fatalf("Unrecognized Limit: %d", options.Limit)
 				}
+			case "":
+				if options.Continue == "C1" {
```

Comment:
what if options.Continue is not "C1". is that an unexpected error case?

Reply:
Yes, it'd be unexpected. Added a check for that.